packages/mui-joy/src/Button/Button.tsx

```diff
@@ -178,16 +223,23 @@ Button.propTypes /* remove-proptypes */ = {
    * Either a string to use a HTML element or a component.
    * This is equivalent to `components.Root`. If both are provided, the `component` is used.
    */
-  component: PropTypes /* @typescript-to-proptypes-ignore */.elementType,
+  component: PropTypes.elementType,
+  /**
+   * The intensity of the shadow.
+   */
+  elevation: PropTypes.oneOf(['lg', 'md', 'sm', 'xl', 'xs']),
   /**
    * If `true`, the button will take up the full width of its container.
    * @default false
    */
   fullWidth: PropTypes.bool,
+  /**
+   * The border-radius of the component.
+   * @default 'default'
+   */
+  roundness: PropTypes.oneOf(['default', 'lg', 'md', 'sm', 'xl', 'xs']),
```

Comment:
I am not sure about `default` as a value. I remember that at some point when we started on v5, we were actually replacing all `default` values for the props that we had. As far as I remember, the reasoning was the the `default` value is not a concrete value, we should rather have it be some specific value, like `md`. @mbrookes what are your thoughts on this?

Reply:
Good point, may be the `roundness` makes sense to not have default because some components can have different values. Ex, Switch might have initial roundness as `xl` whereas Button has 'md'.

Reply:
Exactly, and you can probably use the same argument for each prop, the default for one component may not be the same as the default for another, so I wouldn't add it as a key.

Reply:
Agree that "the default" should be a given size available in the scale (e.g `md` or `sm`) instead of `default` being a plus one value. I think this is a general approach that should be applied to all props, not only border radius, though!

Reply:
Do we really need a prop for this? Is it common for a UI to have multiple buttons with different border radiuses. It seems like it should be configured through the theme. There might be an argument to be made for a "shape" prop (square, rounded, round), but there's a lot to be said for not adding props prematurely.

Reply:
The initial intention for having `shadow` and `radius` props for all components is for API consistency, so the experience of using is the same across components. However, after discussing with Danilo, it is better not to have these props because:
- it encourages dev to make an inconsistent design which is not common for an application to have a different radius for a single component.
- we already have `sx` prop API for one-time customization which can hook to the theme and use the token if needed.
- more props, more maintenance afford.